script/setup/bower

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+set -e
+
+if [ -z "$(which bower)" ]; then
+  echo "A local bower installation is required for bower development." >&2
+  exit 127
+fi
+
+# setup test fixtures
+BASE_PATH="$(cd "$(dirname "${BASH_SOURCE[0]}")/../.." && pwd)"
+cd $BASE_PATH/test/fixtures/bower
+bower install
```

Comment:
It's semantics, but I'd expect `script/bootstrap` to install all dependencies with `script/setup` creating fixtures, getting the DB into the proper state, etc.

> script/bootstrap is used solely for fulfilling dependencies of the project.
>
> This can mean RubyGems, npm packages, Homebrew packages, Ruby versions, Git submodules, etc.
>
> The goal is to make sure all required dependencies are installed.

Specifically, would it make sense to rename the `script/setup/*` individual files to `script/bootstrap/*` files, and call `rake setup` (or `rake bootstrap`) from `script/bootstrap?

Reply:
> It's semantics, but I'd expect script/bootstrap to install all dependencies with script/setup creating fixtures

These setup scripts are creating fixtures ☝️ 

I'd expect bootstrap scripts to be installing tools that are required (cabal/ghc, go, etc) for licensed development, which is outside what the setup scripts are doing.

Reply:
:+1: That makes total sense. Thanks for talking through it with me.

Reply:
Similar to the above, I might add a quick `script/setup` script as well, (even if it just runs `rake setup`).

Reply:
@benbalter thanks for the feedback.  I had to rename the existing `script/setup` folder to not clash with the new script, and settled on `script/source-setup`.  Do you know of any naming precedence that might apply here?

I also thought about naming all the scripts like `script/setup-<source>` but 🤷‍♂️ I liked the folder better for nesting